### PR TITLE
Added new registry adapter for RHCC

### PR DESCRIPTION
**Pull request by dymurray**

### CODE REVIEW

**Comment by jmrodri** [comment]
Run `gofmt -d pkg` and `gofmt -d cmd`. If it prints out a diff, then please fix it. They should return nothing if the format is ok.

File: pkg/apb/rhcc_registry.go [hunk 1]
```diff
@@ -19,5 +40,60 @@ func (r *RHCCRegistry) Init(config RegistryConfig, log *logging.Logger) error {
 // LoadSpecs - Load Red Hat Container Catalog specs
 func (r *RHCCRegistry) LoadSpecs() ([]*Spec, int, error) {
```
Comment (shawn-hurley) [inline pkg/apb/rhcc_registry.go]: If there are no changes to the receiver I think it is more clear if this is not a pointer receiver.

**Comment by tsanders-rh** [comment]
Dylan - Two comments:

1. Should the string we look for be "-apb" vs just "apb".  Wondering if that would pull back less false positive hits initially.

2. Should we make the search string configurable vs hard-coded?

-Todd

**Comment by dymurray** [comment]
@tsanders-rh I initially tried '-apb' by testing with '-rhel' and found that not all images published with "rhel" in the name were showing up. I'm not sure what in the search functionality is causing that but I figured once we are able to test against the real registry I can change the keyword to whatever that gives us the best results. I'm not certain if '-apb' will work or not just yet.

I will change the search string to be configurable though.

**Comment by tsanders-rh** [comment]
Gotcha.  Makes sense.   Thanks!

File: pkg/apb/rhcc_registry.go (hunk 1 above)
Comment (dymurray) [inline pkg/apb/rhcc_registry.go]: Good call, updated this function along with a couple others to reflect that change.

**Comment by shawn-hurley** [comment]
ACK

**Comment by jmrodri** [comment]
configured via `rhcc.config.yaml`, saw it connect:

```
$ make run
go install -ldflags="-s -w" ./cmd/broker
============================================================
==           Starting Ansible Service Broker...           ==
============================================================
[2017-06-05T22:25:02.667-04:00] [DEBUG] Connecting Dao
[2017-06-05T22:25:02.667-04:00] [INFO] == ETCD CX ==
[2017-06-05T22:25:02.667-04:00] [INFO] EtcdHost: localhost
[2017-06-05T22:25:02.667-04:00] [INFO] EtcdPort: 2379
[2017-06-05T22:25:02.667-04:00] [INFO] Endpoints: [http://localhost:2379]
[2017-06-05T22:25:02.668-04:00] [INFO] Etcd Version [Server: 3.1.3, Cluster: 3.1.0]
[2017-06-05T22:25:02.668-04:00] [DEBUG] Creating Cluster Client
[2017-06-05T22:25:02.668-04:00] [WARNING] Failed to create a InternalClientSet: unable to load in-c
luster configuration, KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT must be defined.
[2017-06-05T22:25:02.668-04:00] [DEBUG] Checking for a local Cluster Config
[2017-06-05T22:25:02.674-04:00] [INFO] Cluster Client Created
[2017-06-05T22:25:02.674-04:00] [DEBUG] Connecting to Cluster
[2017-06-05T22:25:02.694-04:00] [INFO] Kubernetes version: v1.5.2+43a9be4
[2017-06-05T22:25:02.694-04:00] [DEBUG] Connecting Registry
[2017-06-05T22:25:02.694-04:00] [INFO] == REGISTRY CX ==
[2017-06-05T22:25:02.694-04:00] [INFO] Name: rhcc
[2017-06-05T22:25:02.694-04:00] [INFO] Url: http://localhost:1337
[2017-06-05T22:25:02.694-04:00] [DEBUG] RHCCRegistry::Init
[2017-06-05T22:25:02.694-04:00] [DEBUG] Initializing WorkEngine
[2017-06-05T22:25:02.694-04:00] [DEBUG] Initializing Provision WorkSubscriber
[2017-06-05T22:25:02.694-04:00] [DEBUG] Creating AnsibleBroker
[2017-06-05T22:25:02.694-04:00] [NOTICE] Ansible Service Broker Started
[2017-06-05T22:25:02.694-04:00] [NOTICE] Listening on http://0.0.0.0:1338
[2017-06-05T22:25:02.695-04:00] [INFO] Listening for provision messages
[2017-06-05T22:25:31.449-04:00] [INFO] AnsibleBroker::Bootstrap
[2017-06-05T22:25:31.45-04:00] [DEBUG] RHCCRegistry::LoadSpecs
[2017-06-05T22:25:31.45-04:00] [DEBUG] RHCCRegistry::LoadImages
[2017-06-05T22:25:31.45-04:00] [DEBUG] Using registry.access.redhat.com to source APB images using
query:apb
[2017-06-05T22:25:32.568-04:00] [DEBUG] Got Image Response from RHCC
[2017-06-05T22:25:32.569-04:00] [DEBUG] Properly unmarshalled image response
[2017-06-05T22:25:32.569-04:00] [DEBUG] Found 0 images in RHCC
::1 - - [05/Jun/2017:22:25:31 -0400] "POST /v2/bootstrap HTTP/1.1" 200 42
 ```

**Comment by fabianvf** [comment]
[(5) RHCC Integration - Implement go-client code to talk to RHCC to source APBs[asb]](https://trello.com/c/Ebx11vsc/233-5-rhcc-integration-implement-go-client-code-to-talk-to-rhcc-to-source-apbsasb)

**Pull request merged**